app/models/mixins/assignment_mixin.rb

```diff
@@ -81,7 +85,13 @@ def get_assigned_tos
         tag = Tag.find_by(:name => "/" + parts.join("/"))
         result[:tags] << [Classification.find_by(:tag_id => tag.id), klass] unless tag.nil?
       when :label
-        label = CustomAttribute.find_by(:name => parts[1], :value => parts[2])
+        label = if parts[1].starts_with?(BASE64_PREFIX)
+                  name = Base64.decode64(parts[1].slice(BASE64_PREFIX.length + 2..-2))
+                  value = Base64.decode64(parts[2].slice(BASE64_PREFIX.length + 2..-2))
```

Comment:
The encoding uses `base64:{...}`, + 2 skips colon and opening brace, -2 stops before closing brace.
I think this would be easier to understand (and test) if you extract the encoding / decoding to separate functions close to each other.

- It'd be more robust for parsing to require colon and braces (eg. with a regexp).  As it is, the simple text "base64isbelongtous" would be considered base64 (and succesfully decoded).
- Don't you want `Base64.strict_decode()`?
- Can labels or custom attributes have `base64:{...}` form and trick us into decoding where we shouldn't have decoded?
  - It may be better to disambiguate out-of-band, eg. by using `/label_base64/...` namespace vs `/label/...`?
- Either way, should assume decoding can raise exception.